### PR TITLE
Reland exposer

### PR DESCRIPTION
**Pull request by djmitche**
This relands #166 and #176.  Tests on master aren't passing for me locally, so let's see how they do in CI..

### CODE REVIEW

**Comment by djmitche** [comment]
The windows test failed for what looks like an unrelated reason.  I've re-run it.

**Comment by djmitche** [comment]
All green in TC :)

**Comment by djmitche** [comment]
BTW, I ran the linux tests in a loop locally and then forgot about it, so it ran some many thousands of time without error.

File: livelog.go [hunk 1]
```diff
@@ -153,30 +157,34 @@ func (l *LiveLogTask) reinstateBackingLog() {
...
+	exposeURL := l.exposure.GetURL()
+	if exposeURL.Path == "/" {
+		exposeURL.Path = logURL.Path
 	} else {
-		getURL.Scheme = "http"
+		exposeURL.Path = exposeURL.Path + logURL.Path
 	}
-	getURL.Host = statelessHostname + ":" + strconv.Itoa(int(l.liveLog.GETPort))
+
+	expires := time.Now().Add(time.Duration(l.task.Payload.MaxRunTime+900) * time.Second)
```
Comment (petemoore) [inline livelog.go]: a comment to explain that the 900 is to account for clock drift would be useful here

File: livelog/livelog.go [hunk 2]
```diff
@@ -21,14 +21,11 @@ import (
 // LiveLog provides access to a livelog process running on the OS. Use
 // New(liveLogExecutable string) to start a new livelog instance.
 type LiveLog struct {
-	SSLCert string
-	SSLKey  string
-	secret  string
-	PUTPort uint16
-	GETPort uint16
-	putURL  string
-	// The fully qualified HTTP GET URL where the log will be published.
```
Comment (petemoore) [inline livelog/livelog.go]: Did you intend to delete this comment for `GetURL`?

**Pull request merged**